pkg/collect/host_certificate.go

```diff
@@ -77,3 +116,66 @@ func isEncryptedKey(filename string) (bool, error) {
 	}
 	return bytes.Contains(data, []byte("ENCRYPTED")), nil
 }
+
+func HostCertParser(certPath string) HostCertCollection {
+	cert, err := ioutil.ReadFile(certPath)
+	var certInfo []HostParsedCertificate
+	if err != nil {
+		return HostCertCollection{
+			CertificatePath: certPath,
+			Message:         CertMissing,
+		}
+	}
+
+	certChain := HostDecodePem(cert)
+
+	if len(certChain.Certificate) == 0 {
+		return HostCertCollection{
+			CertificatePath: certPath,
+			Message:         CertInvalid,
+		}
+	}
+
+	for _, cert := range certChain.Certificate {
+		parsedCert, errParse := x509.ParseCertificate(cert)
+		if errParse != nil {
+			return HostCertCollection{
+				CertificatePath: certPath,
+				Message:         CertInvalid,
+			}
+		}
+		currentTime := time.Now()
+		certInfo = append(certInfo, HostParsedCertificate{
+			Subject:                 parsedCert.Subject.ToRDNSequence().String(),
+			SubjectAlternativeNames: parsedCert.DNSNames,
+			Issuer:                  parsedCert.Issuer.ToRDNSequence().String(),
+			NotAfter:                parsedCert.NotAfter,
+			NotBefore:               parsedCert.NotBefore,
+			IsExpired:               currentTime.After(parsedCert.NotAfter),
+			IsCA:                    parsedCert.IsCA,
+		})
+	}
+
+	return HostCertCollection{
+		CertificatePath:  certPath,
+		CertificateChain: certInfo,
+		Message:          CertValid,
+	}
+}
+
+func HostDecodePem(certInput []byte) tls.Certificate {
```

Comment:
No need to duplicate this code. Just use `decodePem`. If you are concerned about `decodePem` function being part of `collect/collect.go`, you can move it to `collect/util.go` like other shared functions.

Reply:
Thank you! I will move it to collect/util.go